registry/client/errors.go

```diff
@@ -37,30 +37,25 @@ func (e *UnexpectedHTTPResponseError) Error() string {
 	return fmt.Sprintf("error parsing HTTP %d response body: %s: %q", e.StatusCode, e.ParseErr.Error(), string(e.Response))
 }
 
-func parseHTTPErrorResponse(statusCode int, r io.Reader) error {
+func parseHTTPErrorResponse(statusCode int, r io.Reader, contentType string) error {
 	var errors errcode.Errors
 	body, err := io.ReadAll(r)
 	if err != nil {
 		return err
 	}
 
+	if contentType != "application/json" && contentType != "application/vnd.api+json" {
```

Comment:
This may need additional handling; the content-type may have additional fields (e.g. charset) after it, see [RFC9110, section 8.3](https://www.rfc-editor.org/rfc/rfc9110#field.content-type); also se another PR related to that;

- https://github.com/distribution/distribution/pull/1363

Reply:
I'm still looking if there's some canonical utility for that in stdlib, but I don't think there is. (same for `Accept` headers, which are even more horrible to parse)

Reply:
I used `mime.ParseMediaType` - I think it's the best tool for the job 😄

Reply:
Oh! You're right, that one's there indeed 👍 I think it was the `Accept` header that didn't have a good one in stdlib (to handle the `q` / priority / weight